chains/solana/contracts/programs/burnmint-token-pool/src/lib.rs

```diff
@@ -19,45 +19,78 @@ pub mod burnmint_token_pool {
 
     use super::*;
 
-    pub fn init_global_config(ctx: Context<InitGlobalConfig>) -> Result<()> {
+    pub fn init_global_config(
+        ctx: Context<InitGlobalConfig>,
+        router_address: Pubkey,
+        rmn_address: Pubkey,
+    ) -> Result<()> {
         ctx.accounts.config.set_inner(PoolConfig {
             self_served_allowed: false,
             version: 1,
+            router: router_address,
+            rmn_remote: rmn_address,
         });
 
         emit!(GlobalConfigUpdated {
             self_served_allowed: ctx.accounts.config.self_served_allowed,
+            router: ctx.accounts.config.router,
+            rmn_remote: ctx.accounts.config.rmn_remote,
         });
 
         Ok(())
     }
 
-    pub fn update_global_config(
+    pub fn update_self_served_allowed(
         ctx: Context<UpdateGlobalConfig>,
         self_served_allowed: bool,
     ) -> Result<()> {
         ctx.accounts.config.self_served_allowed = self_served_allowed;
 
         emit!(GlobalConfigUpdated {
             self_served_allowed: ctx.accounts.config.self_served_allowed,
+            router: ctx.accounts.config.router,
+            rmn_remote: ctx.accounts.config.rmn_remote,
         });
 
         Ok(())
     }
 
-    pub fn initialize(
-        ctx: Context<InitializeTokenPool>,
-        router: Pubkey,
-        rmn_remote: Pubkey,
+    pub fn update_default_router(
+        ctx: Context<UpdateGlobalConfig>,
+        router_address: Pubkey,
     ) -> Result<()> {
+        ctx.accounts.config.router = router_address;
+
+        emit!(GlobalConfigUpdated {
+            self_served_allowed: ctx.accounts.config.self_served_allowed,
+            router: ctx.accounts.config.router,
+            rmn_remote: ctx.accounts.config.rmn_remote,
+        });
+
+        Ok(())
+    }
+
+    pub fn update_default_rmn(ctx: Context<UpdateGlobalConfig>, rmn_address: Pubkey) -> Result<()> {
+        ctx.accounts.config.rmn_remote = rmn_address;
+
+        emit!(GlobalConfigUpdated {
+            self_served_allowed: ctx.accounts.config.self_served_allowed,
+            router: ctx.accounts.config.router,
+            rmn_remote: ctx.accounts.config.rmn_remote,
+        });
+
+        Ok(())
+    }
+
+    pub fn initialize(ctx: Context<InitializeTokenPool>) -> Result<()> {
```

Comment:
I think you're already working on it, but just in case: The removal of the two Pubkeys in the instruction data will require changes in tooling, probably e2e tests, and should be accounted for in the CCT docs (whenever those are available anyway)